### PR TITLE
USHIFT-3459: Passthrough kubelet config from microshift config

### PR DESCRIPTION
**Pull request by pmtk**

### CODE REVIEW

**Review by copejon — COMMENTED** [review]
A couple non-blocker nits, otherwise lgtm!

File: pkg/node/kubelet.go [hunk 1]
```diff
@@ -112,25 +127,31 @@ func (s *KubeletServer) writeConfig(cfg *config.Config) error {
 		resolvConf = config.DefaultSystemdResolvedFile
 	}
 
+	passthrough := ""
```
[the remark this answers is not in the document]
Comment (pmtk) [reply pkg/node/kubelet.go:130]: Renamed to `userProvidedConfig`, wdyt?

File: docs/user/howto_config.md [hunk 2]
```diff
@@ -91,6 +92,7 @@ ingress:
     routeAdmissionPolicy:
         namespaceOwnership: InterNamespaceAllowed
     status: Managed
+kubelet:
```
[the remark this answers is not in the document]
Comment (pmtk) [reply docs/user/howto_config.md:95]: I agree that an example would be nice, but it seems very manual process compared to what we have now. Also, current examples are showing default values, which still holds true for current `kubelet:`.

I don't see elegant way to do this, or a way to not confuse users, but we'll definitely explain this feature in the official docs. 
WDYT, can we skip this one?

**Comment by pmtk** [comment]
/hold
Let's think if we'd be better using new kubelet's drop-in dir cfg

File: docs/user/howto_config.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (pacevedom) [reply docs/user/howto_config.md:95]: can you force it to have a `null` value? This makes it to the default config and we need to be able to load it without issues.

File: pkg/config/config.go [hunk 3]
```diff
@@ -48,6 +48,10 @@ type Config struct {
 	Manifests Manifests     `json:"manifests"`
 	Ingress   IngressConfig `json:"ingress"`
 
+	// Settings specified in this section are transferred as-is into the Kubelet config.
+	// +kubebuilder:validation:Schemaless
+	Kubelet map[string]any `json:"kubelet"`
```
Comment (pacevedom) [inline pkg/config/config.go:53]: `// +kubebuilder:default=null`
Given that you already have schemaless you can put whatever you want here, meaning `null` should work.

File: docs/user/howto_config.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (pmtk) [reply docs/user/howto_config.md:95]: Empty object such as example below is a valid yaml AFAIK, is it not?
```
kubelet:
```

[the remark this answers is not in the document]
Comment (pmtk) [reply docs/user/howto_config.md:95]: I double checked and `kubelet:` loads just fine and personally I prefer it to `kubelet: null`

**Review by pacevedom — APPROVED** [review]
/lgtm

**Pull request merged**